framework/sandstone_p.h

```diff
@@ -334,6 +334,7 @@ struct SandstoneApplication : public InterruptMonitor, public test_the_test_data
     Duration delay_between_tests = std::chrono::milliseconds(5);
 
     std::unique_ptr<RandomEngineWrapper, RandomEngineDeleter> random_engine;
+    bool random_compatibility_mode = false;
```

Comment:
Please find a group of `bool` to move this next to. There's a 1-byte padding hole right now near `ud_on_failure` and `service_background_scan`.

Additionally, this variable seems to be needed in the exec state of the child process, so please add it to the `APP_STATE_VARIABLES` macro.